flow/view_holder.cc

```diff
@@ -0,0 +1,121 @@
+  view_holder_token_.value.reset(view_holder_token_handle->ReleaseHandle());
+}
+
+ViewHolder::~ViewHolder() {
+  // Ensure that we properly released the resource.
+  FML_DCHECK(!view_holder_);
+  FML_DCHECK(scene_update_context_ == nullptr);
+}
+
+void ViewHolder::Bind(SceneUpdateContext& context,
+                      scenic::ContainerNode& container,
+                      const SkPoint& offset,
+                      bool hit_testable) {
+  FML_LOG(ERROR) << "ViewHolder::Bind";
+  if (view_holder_token_.value) {
+    FML_LOG(ERROR) << "CREATE NODES";
+    // Happens first time we bind.
+    // The Resource ID message MUST be posted before the resource is created, or
+    // race conditions could occur!
+    scenic::ResourceId view_holder_id = container.session()->next_resource_id();
+    on_bind_callback_(view_holder_id);
+
+    // Create resources.  This might cause an immediate session flush, so this
+    // must be done after registering the view_holder_id via the callback.
+    view_holder_.reset(new scenic::ViewHolder(
+        container.session(), std::move(view_holder_token_.value),
+        "Flutter SceneHost"));
+    transform_node_.reset(new scenic::EntityNode(container.session()));
+    transform_node_->Attach(*view_holder_);
+
+    scene_update_context_ = &context;
+  }
+
+  if (view_holder_) {
+    FML_LOG(ERROR) << "ATTACH NODES";
+    transform_node_->SetTranslation(offset.x(), offset.y(), 0.f);
+    transform_node_->SetHitTestBehavior(
+        hit_testable ? fuchsia::ui::gfx::HitTestBehavior::kDefault
+                     : fuchsia::ui::gfx::HitTestBehavior::kSuppress);
+    container.AddChild(*transform_node_);
+
+    if (pending_properties_) {
+      SetProperties(*pending_properties_.release());
+    }
+  }
+}
+
+void ViewHolder::Dispose(bool remove_from_scene_update_context) {
+  FML_LOG(ERROR) << "ViewHolder::Dispose";
+  // If scene_update_context_ is set, then we should still have a ViewHolder
+  // left to dereference.
+  // If scene_update_context_ is null, then either:
+  // 1. A ViewHolder was never created, or
+  // 2. A ViewHolder was created but was already dereferenced (i.e. Dispose has
+  // already been called).
+  FML_DCHECK(scene_update_context_ || !view_holder_);
+
+  if (remove_from_scene_update_context && scene_update_context_) {
+    scene_update_context_->RemoveViewHolder(this);
+  }
+  if (view_holder_ != nullptr) {
+    on_unbind_callback_(view_holder_->id());
+  }
+
+  view_holder_token_.value.reset();
+  scene_update_context_ = nullptr;
+  transform_node_ = nullptr;
+  view_holder_ = nullptr;
+}
+
+void ViewHolder::SetProperties(
+    fuchsia::ui::gfx::ViewProperties new_properties) {
+  if (!view_holder_) {
+    pending_properties_ =
+        std::make_unique<fuchsia::ui::gfx::ViewProperties>(new_properties);
+    return;
+  }
+
+  // TODO: focus change
+  FML_LOG(ERROR) << "ViewHolder::SetProperties";
+  view_holder_->SetViewProperties(new_properties);
+}
+
+ViewHolderProxy::ViewHolderProxy(
```

Comment:
Are there other things in Flutter named Holder besides ExportNodeHolder.  If this is a common naming pattern, it's probably better to say ViewHolderHolder rather than to "innovate" with this naming.  Or perhaps rename "ViewHolder" to "EmbeddedView".

Reply:
I ended up refactoring the code to remove this split entirely, so the issue with the naming went away too